### ISSUE TITLE
The official compiler snapshots dump a bunch of *.emit-module.d files in the package directory

### ISSUE THREAD

**Issue by buttaface** [apple/swift#58405]
**Describe the bug**
I've been seeing a bunch of emit-module.d files in my Swift package directories with the latest 5.7 and trunk snapshot builds. The first Apr. 12 5.7 snapshot doesn't have this problem, so building with `swift build -v` then comparing to the Apr. 18 snapshot shows that both run this identical command to emit a module:
```
/home/butta/swift-nio/swift-5.7-DEVELOPMENT-SNAPSHOT-2022-04-18-a-ubuntu20.04/usr/bin/swiftc -module-name _NIODataStructures -incremental -emit-dependencies -emit-module -emit-module-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/_NIODataStructures.swiftmodule -output-file-map /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/_NIODataStructures.build/output-file-map.json -parse-as-library -c /home/butta/swift-nio/Sources/_NIODataStructures/Heap.swift /home/butta/swift-nio/Sources/_NIODataStructures/PriorityQueue.swift -I /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug -target x86_64-unknown-linux-gnu -swift-version 5 -enable-batch-mode -index-store-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/index/store -Onone -enable-testing -g -j1 -DSWIFT_PACKAGE -DDEBUG -module-cache-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/ModuleCache -parseable-output -parse-as-library -v
```
but when translated to the `swift-frontend` only the later snapshot adds an unprefixed `-emit-dependencies-path`:
```
Swift version 5.7-dev (LLVM 8bbf73b53c59eeb, Swift a159921d1bf3336)Target: x86_64-unknown-linux-gnu/home/butta/swift-nio/swift-5.7-DEVELOPMENT-SNAPSHOT-2022-04-18-a-ubuntu20.04/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /home/butta/swift-nio/Sources/_NIODataStructures/Heap.swift /home/butta/swift-nio/Sources/_NIODataStructures/PriorityQueue.swift -target x86_64-unknown-linux-gnu -disable-objc-interop -I /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug -enable-testing -g -module-cache-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/ModuleCache -swift-version 5 -Onone -D SWIFT_PACKAGE -D DEBUG -new-driver-path /home/butta/swift-nio/swift-5.7-DEVELOPMENT-SNAPSHOT-2022-04-18-a-ubuntu20.04/usr/bin/swift-driver -empty-abi-descriptor -resource-dir /home/butta/swift-nio/swift-5.7-DEVELOPMENT-SNAPSHOT-2022-04-18-a-ubuntu20.04/usr/lib/swift -enable-anonymous-context-mangled-names -module-name _NIODataStructures -emit-module-doc-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/_NIODataStructures.swiftdoc -emit-module-source-info-path /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/_NIODataStructures.swiftsourceinfo -emit-dependencies-path _NIODataStructures.emit-module.d -parse-as-library -o /home/butta/swift-nio/.build/x86_64-unknown-linux-gnu/debug/_NIODataStructures.swiftmodule
```
which causes all those files to be dumped in the top level of the package, rather than in `.build/` where they belong. I can reproduce with the Apr. 20 trunk snapshot and later for Ubuntu too. Adding `-Xswiftc -disallow-use-new-driver` works around the problem.

**To Reproduce**
1. Install a recent snapshot of the Swift toolchain.
2. Run `swift build` on any Swift package.
3. Check with `ls *.emit-module.d`.

**Expected behavior**
No build files dumped outside the `.build` directory.

**Environment:**
 - OS: Ubuntu 20.04 x86_64

**Additional context**
apple/swift-driver#1056 most likely caused this problem.

**Comment by karwa** [comment]
I'm seeing this as well on macOS, using the snapshot via Xcode.

![image](https://user-images.githubusercontent.com/5254025/165975444-594911bf-83ce-4959-8c13-57c6e6186a50.png)

**Comment by buttaface** [comment]
These files are now placed in `.build/` after @compnerd's fix in apple/swift-driver#1088, checked with the May. 4th trunk snapshot build.

**Comment by buttaface** [comment]
Just downloaded the May 10th snapshot of the 5.7 toolchain and it doesn't have the problem either, now that that fix was backported in apple/swift-driver#1094.
